contracts/Vault.vy

```diff
@@ -553,36 +556,40 @@ def setWithdrawalQueue(queue: address[MAXIMUM_STRATEGIES]):
     """
     assert msg.sender in [self.management, self.governance]
 
-    # HACK: Temporary until Vyper adds support for Dynamic arrays
-    old_queue: address[MAXIMUM_STRATEGIES] = empty(address[MAXIMUM_STRATEGIES])
+    set: address[SET_SIZE] = empty(address[SET_SIZE])
     for i in range(MAXIMUM_STRATEGIES):
-        old_queue[i] = self.withdrawalQueue[i] 
         if queue[i] == ZERO_ADDRESS:
             # NOTE: Cannot use this method to remove entries from the queue
-            assert old_queue[i] == ZERO_ADDRESS
+            assert self.withdrawalQueue[i] == ZERO_ADDRESS
             break
         # NOTE: Cannot use this method to add more entries to the queue
-        assert old_queue[i] != ZERO_ADDRESS
+        assert self.withdrawalQueue[i] != ZERO_ADDRESS
 
         assert self.strategies[queue[i]].activation > 0
 
-        existsInOldQueue: bool = False
-        for j in range(MAXIMUM_STRATEGIES):
-            if queue[j] == ZERO_ADDRESS:
-                existsInOldQueue = True
+        # hash will have hash of the address in uint256.
+        # `mask = SET_SIZE - 1` will be used to get key for the set for indexing
+        hash: uint256 = convert(queue[i], uint256)
+        key: uint256 = bitwise_and(hash, SET_SIZE - 1)  # Key is first `SET_SIZE` bits of address
+        # Most of the times following for loop only run once which is making it highly gas efficient 
+        # but in the worst case of key collision it will run linearly and find first empty slot.
+        for m in range(key, key + SET_SIZE):
+            if m >= SET_SIZE:
+                for n in range(SET_SIZE):
+                    # Note: This only means SET_SIZE is lower than MAXIMUM_STRATEGIES
+                    assert n < key
```

Comment:
Delete this line to save a tiny amount of gas now that we raise if the set is filled up prematurely
```suggestion
```

Reply:
Yes, this will waste a little extra gas getting to the final conclusion, but this invariant will still be maintained by proxy, and also it reduces costs in the common case workflow of 1-2 passes before filling in the set